datastream-bigquery/src/main/java/com/linkedin/datastream/bigquery/BigqueryTransportProvider.java

```diff
@@ -5,184 +5,237 @@
+        _destinationConfigurations = destinationConfigurations;
+        _destinations = new HashSet<>();
     }
 
     @Override
-    protected void shutdownCommitter() {
-        _committer.shutdown();
+    public void send(final String destination, final DatastreamProducerRecord record, final SendCallback onComplete) {
+        final BigqueryDatastreamDestination datastreamDestination = BigqueryDatastreamDestination.parse(destination);
+        registerConfigurationForDestination(datastreamDestination, _datastreamConfiguration);
+        final SendCallback callbackHandler = _datastreamConfiguration.getDeadLetterTableConfiguration()
+                .map(deadLetterTableConfiguration -> exceptionHandlingCallbackHandler(datastreamDestination, record, onComplete, deadLetterTableConfiguration))
+                .orElse(onComplete);
+        _bufferedTransportProvider.send(destination, record, callbackHandler);
     }
 
-    /**
-     * Builder class for {@link com.linkedin.datastream.bigquery.BigqueryTransportProvider}
-     */
-    public static class BigqueryTransportProviderBuilder {
-        private String _transportProviderName;
-        private int _batchBuilderQueueSize;
-        private int _batchBuilderCount;
-        private int _maxBatchSize;
-        private int _maxBatchAge;
-        private int _maxInflightBatchCommits;
-        private BigqueryBatchCommitter _committer;
-        private VerifiableProperties _translatorProperties;
-        private Map<String, BigquerySchemaEvolver> schemaEvolvers;
-        private String defaultSchemaEvolverName;
-
-        /**
-         * Set the name of the transport provider
-         */
-        public BigqueryTransportProviderBuilder setTransportProviderName(String transportProviderName) {
-            this._transportProviderName = transportProviderName;
-            return this;
-        }
-
-        /**
-         * Set batch builder's queue size
-         */
-        public BigqueryTransportProviderBuilder setBatchBuilderQueueSize(int batchBuilderQueueSize) {
-            this._batchBuilderQueueSize = batchBuilderQueueSize;
-            return this;
-        }
-
-        /**
-         * Set number of batch builders
-         */
-        public BigqueryTransportProviderBuilder setBatchBuilderCount(int batchBuilderCount) {
-            this._batchBuilderCount = batchBuilderCount;
-            return this;
-        }
+    private SendCallback exceptionHandlingCallbackHandler(final BigqueryDatastreamDestination datastreamDestination,
+                                                          final DatastreamProducerRecord record,
+                                                          final SendCallback onComplete,
+                                                          final BigqueryDatastreamConfiguration config) {
+        return (metadata, exception) -> {
+            if (exception == null || exception instanceof DatastreamTransientException) {
```

Comment:
I haven’t gone through rest of the code. So I am assuming a bit here.

Please leave a comment why DatastreamTransientException is special and it doesn't necessarily mean schema evolution errors.

Reply:
The DatastreamTransientException exception indicates that there was a temporary issue writing the record to BigQuery, e.g. an error connecting to GCP, so the record should be retried via the built-in Brooklin retry logic. For exceptions that are not DatastreamTransientException, we consider these to be unrecoverable exceptions, in which case we route the record to the exceptions/dead-letter table.

Reply:
I understand that. I guess what I am saying is, can we have that explanation in the code as a comment?
Just trying to give context inline to someone new to the code.